src/lv_core/lv_indev.c

```diff
@@ -1051,39 +1055,20 @@ static void indev_drag(lv_indev_proc_t * state)
             /*Get the coordinates of the object and modify them*/
             lv_coord_t act_x      = lv_obj_get_x(drag_obj);
             lv_coord_t act_y      = lv_obj_get_y(drag_obj);
-            uint16_t inv_buf_size = lv_disp_get_inv_buf_size(
-                indev_act->driver.disp); /*Get the number of currently invalidated areas*/
-
-            lv_coord_t prev_x     = drag_obj->coords.x1;
-            lv_coord_t prev_y     = drag_obj->coords.y1;
-            lv_coord_t prev_par_w = lv_obj_get_width(lv_obj_get_parent(drag_obj));
-            lv_coord_t prev_par_h = lv_obj_get_height(lv_obj_get_parent(drag_obj));
-
-            lv_obj_set_pos(drag_obj, act_x + state->types.pointer.vect.x,
-                           act_y + state->types.pointer.vect.y);
-
-            /*Set the drag in progress flag if the object is really moved*/
-            if(drag_obj->coords.x1 != prev_x || drag_obj->coords.y1 != prev_y) {
-                if(state->types.pointer.drag_in_prog !=
-                   0) { /*Send the drag begin signal on first move*/
-                    drag_obj->signal_cb(drag_obj, LV_SIGNAL_DRAG_BEGIN, indev_act);
-                    if(state->reset_query != 0) return;
-                }
-                state->types.pointer.drag_in_prog = 1;
-            }
-            /*If the object didn't moved then clear the invalidated areas*/
-            else {
-                /*In a special case if the object is moved on a page and
-                 * the scrollable has fit == true and the object is dragged of the page then
-                 * while its coordinate is not changing only the parent's size is reduced */
-                lv_coord_t act_par_w = lv_obj_get_width(lv_obj_get_parent(drag_obj));
-                lv_coord_t act_par_h = lv_obj_get_height(lv_obj_get_parent(drag_obj));
-                if(act_par_w == prev_par_w && act_par_h == prev_par_h) {
-                    uint16_t new_inv_buf_size = lv_disp_get_inv_buf_size(indev_act->driver.disp);
-                    lv_disp_pop_from_inv_buf(indev_act->driver.disp,
-                                             new_inv_buf_size - inv_buf_size);
```

Comment:
This code (or similar) should remain because it prevents to invalidate the areas if the object was not moving. (Tried to drag but reach e.g. the edge of a page.)

Reply:
Hmm, alright, I'll have to investigate that. The problem I had was that the screen wasn't refreshing properly at all when scrolling with this code enabled.

Reply:
The issue I experienced appears to have been a typo on my end. Please disregard my last message. The optimization has been added back.